### DISCUSSION TITLE
add Ignore parameter on TableColumn

### DISCUSSION THREAD

**Discussion by A5196060** [dotnetcore/BootstrapBlazor#3366] [Q&A] [answered]
### `TableColumn` 缺少 `Ignore` 参数

`AutoGenerateBaseAttribute` 有参数 `Ignore` 负责将模型 `Model` 中的属性自动忽略掉

`TableColumn` 组件中没有相对应的参数来控制

### 应用场景

表格需要一个按钮来控制列是否显示，通过 `Visible` 控制是否显示后 `ColumnList` 内始终显示这列，并且可以勾选重新显示此列，缺少一个即不渲染也不显示在 `ColumnList` 的选项，所以需要增加一个 `Ignore` 来实现此功能

`Visible` 应用示例
https://www.blazor.zone/table/column/list#Visible

`Ignore` 应用示例
https://www.blazor.zone/table/column#Ignore

### 属性混乱

目前有 `Editable` `Visible` `IsVisibleWhenAdd` `IsVisibleWhenEdit` `Readonly` `IsReadonlyWhenAdd` `IsReadonlyWhenEdit`

**Comment by ArgoZhang** [comment, marked as answer]
### 总结

1. 移除 `Editable`
2. 增加 `Ignore` 控制列是否忽略，`Ignore="true"` 后所有场景中均看不到此列信息，新建、编辑、表格显示、`ColumnList` 中均不出现
3. `Readonly` 控制模型属性或者字段是否可以编辑，未设置 `IsReadonlyWhenAdd` `IsReadonlyWhenEdit` 时以 `Readonly` 值为准，设置 `IsReadonlyWhenAdd` 值时新建时以此值为准，设置 `IsReadonlyWhenEdit` 值时编辑时以此值为准
4. `Visible` 控制模型属性或者字段是否可见，`IsVisibleWhenAdd` `IsVisibleWhenEdit` 时以 `Visible` 值为准，设置 `IsVisibleWhenAdd` 值时新建时以此值为准，设置 `IsVisibleWhenEdit` 值时编辑时以此值为准

### 接口更新

`IEditorItem` 对标 `EditorForm` 更新如下
1. 移除 `Editable` `Visible` `IsReadonlyWhenAdd` `IsReadonlyWhenEdit` `IsVisibleWhenAdd` `IsVisibleWhenEdit`
2. 增加 `Ignore`

`ITableColumn` 对标 `TableColumn` 更新如下
1. 增加 `Visible` `IsReadonlyWhenAdd` `IsReadonlyWhenEdit` `IsVisibleWhenAdd` `IsVisibleWhenEdit`

`IEditorItem` 应该有 `Ignore` `Readonly`
`ITableColumn` 增加 `Visible`
